daemon/containerd/service.go

```diff
@@ -99,12 +104,53 @@ func (i *ImageService) ReleaseLayer(rwlayer layer.RWLayer) error {
 // LayerDiskUsage returns the number of bytes used by layer stores
 // called from disk_usage.go
 func (i *ImageService) LayerDiskUsage(ctx context.Context) (int64, error) {
-	panic("not implemented")
+	ch := i.usage.DoChan("LayerDiskUsage", func() (interface{}, error) {
+		var allLayersSize int64
+		snapshotter := i.client.SnapshotService(containerd.DefaultSnapshotter)
```

Comment:
Make sure to check if code didn't change after the original PR was merged in the fork. For example, on this one; https://github.com/rumpl/moby/blame/e4cd5b9a89f311b9032a1aba5e9c8c1d4cda544e/daemon/containerd/service.go#L111

Code was later updated in https://github.com/rumpl/moby/commit/99655cc01035ff599216449d1108430a7edeed16, but parts of that already upstreamed, so this code needs to be updated accordingly.


What we should probably do, is do another rebase of https://github.com/moby/moby/pull/43902, do conflict resolution on that branch, and then take the commits after rebasing to upstream.

Reply:
indeed, thanks